python/lance/__init__.py

```diff
@@ -107,11 +107,11 @@ def diff(dataset: FileSystemDataset, v1: int, v2: int = None) -> LanceDiff:
         dataset
     """
     if v1 < 0:
-        v1 = dataset.versions()[v1]["version"]
+        v1 = dataset.versions()[v1-1]["version"]
```

Comment:
Can you explain the logic here?  so `-1` is actually access `versions[-2]`?
means `latest == version[0]`?

Reply:
yeah, it's a bit confusing. Version=0 being latest is the existing code convention. I wasn't thinking about the indexing as a list index but offsets from latest.

Reply:
Lets add them into document?